### ISSUE TITLE
Prettier step is ignored for Solidity files on Mac

### ISSUE THREAD

**Issue by pcaversaccio** [prettier-solidity/prettier-plugin-solidity#978]
For reproduction, please use my repo [here](https://github.com/pcaversaccio/snekmate).

@cairoeth reported to me that invoking the script [`pnpm prettier:fix`](https://github.com/pcaversaccio/snekmate/blob/main/package.json#L26) ignored (important: there is no error raised) the Solidity files. Directly invoking it using `npx prettier -w **/*.{md,sol,js,json,yml,yaml}` however worked. If I add `'...'` to the script, it will fix it for him: `npx prettier -w '**/*.{md,sol,js,json,yml,yaml}'`.

I was not able to reproduce this "error" myself on my Windows nor Ubuntu machine (also all my CI runs using Ubuntu work as intended). @cairoeth uses Mac and he tested the behaviour using zsh and bash. For example, on my Ubuntu machine using bash it worked well. So I'm kinda lost why it behaves like this...

That's the current output `pnpm prettier:fix` produces for @cairoeth on his Mac using zsh:

```console
> snekmate@0.0.5-rc.1 prettier:fix /Users/cairo/Desktop/snekmate
> npx prettier -w **/*.{md,sol,js,json,yml,yaml}

.github/pull_request_template.md 17ms (unchanged)
CHANGELOG.md 35ms (unchanged)
CONTRIBUTING.md 11ms (unchanged)
GUIDELINES.md 15ms (unchanged)
README.md 15ms (unchanged)
SECURITY.md 2ms (unchanged)
.github/dependabot.yml 11ms (unchanged)
.github/FUNDING.yml 1ms (unchanged)
.github/ISSUE_TEMPLATE/bug_report.yml 5ms (unchanged)
.github/ISSUE_TEMPLATE/feature_request.yml 2ms (unchanged)
.github/workflows/checks.yml 7ms (unchanged)
.github/workflows/publish-npm.yml 3ms (unchanged)
.github/workflows/publish-pypi.yml 3ms (unchanged)
.github/workflows/publish-test-pypi.yml 2ms (unchanged)
.github/workflows/test-contracts.yml 5ms (unchanged)
.prettierrc.yml 1ms (unchanged)
ape-config.yaml 1ms (unchanged)
pnpm-lock.yaml 109ms
```

**Comment by Janther** [comment]
This is very weird,

I cloned the repo, and installed the dependencies using `pnpm install` and when I ran `pnpm prettier:fix`, the output is much larger than yours, not only containing `.sol` packages but even `.js` and `.json` files which are also missing in your output.

```zsh
> snekmate@0.0.5-rc.1 prettier:fix /Users/.../snekmate
> npx prettier -w **/*.{md,sol,js,json,yml,yaml}

.github/pull_request_template.md 15ms (unchanged)
CHANGELOG.md 37ms (unchanged)
CONTRIBUTING.md 10ms (unchanged)
GUIDELINES.md 14ms (unchanged)
README.md 15ms (unchanged)
SECURITY.md 2ms (unchanged)
lib/utils/VyperDeployer.sol 103ms (unchanged)
test/auth/AccessControl.t.sol 228ms (unchanged)
test/auth/interfaces/IAccessControlExtended.sol 3ms (unchanged)
test/auth/interfaces/IOwnable.sol 5ms (unchanged)
test/auth/interfaces/IOwnable2Step.sol 1ms (unchanged)
test/auth/Ownable.t.sol 36ms (unchanged)
test/auth/Ownable2Step.t.sol 71ms (unchanged)
test/extensions/ERC2981.t.sol 175ms (unchanged)
test/extensions/ERC4626.t.sol 348ms (unchanged)
test/extensions/interfaces/IERC2981Extended.sol 2ms (unchanged)
test/extensions/interfaces/IERC4626Extended.sol 1ms (unchanged)
test/extensions/mocks/ERC20ExcessDecimalsMock.sol 1ms (unchanged)
test/tokens/ERC1155.t.sol 768ms (unchanged)
test/tokens/ERC20.t.sol 248ms (unchanged)
test/tokens/ERC721.t.sol 414ms (unchanged)
test/tokens/interfaces/IERC1155Extended.sol 1ms (unchanged)
test/tokens/interfaces/IERC20Extended.sol 1ms (unchanged)
test/tokens/interfaces/IERC4494.sol 0ms (unchanged)
test/tokens/interfaces/IERC721Extended.sol 1ms (unchanged)
test/tokens/mocks/ERC1155ReceiverMock.sol 5ms (unchanged)
test/tokens/mocks/ERC721ReceiverMock.sol 5ms (unchanged)
test/utils/Base64.t.sol 50ms (unchanged)
test/utils/BatchDistributor.t.sol 87ms (unchanged)
test/utils/Create2Address.t.sol 18ms (unchanged)
test/utils/CreateAddress.t.sol 124ms (unchanged)
test/utils/ECDSA.t.sol 80ms (unchanged)
test/utils/EIP712DomainSeparator.t.sol 24ms (unchanged)
test/utils/interfaces/IBase64.sol 1ms (unchanged)
test/utils/interfaces/IBatchDistributor.sol 1ms (unchanged)
test/utils/interfaces/ICreate2Address.sol 1ms (unchanged)
test/utils/interfaces/ICreateAddress.sol 0ms (unchanged)
test/utils/interfaces/IECDSA.sol 1ms (unchanged)
test/utils/interfaces/IEIP712DomainSeparator.sol 0ms (unchanged)
test/utils/interfaces/IMath.sol 2ms (unchanged)
test/utils/interfaces/IMerkleProofVerification.sol 1ms (unchanged)
test/utils/interfaces/IMulticall.sol 1ms (unchanged)
test/utils/interfaces/ISignatureChecker.sol 1ms (unchanged)
test/utils/Math.t.sol 143ms (unchanged)
test/utils/MerkleProofVerification.t.sol 118ms (unchanged)
test/utils/mocks/Create2Impl.sol 4ms (unchanged)
test/utils/mocks/ERC1271MaliciousMock.sol 1ms (unchanged)
test/utils/mocks/ERC1271WalletMock.sol 2ms (unchanged)
test/utils/mocks/ERC20Mock.sol 2ms (unchanged)
test/utils/mocks/EtherReceiver.sol 1ms (unchanged)
test/utils/mocks/MockCallee.sol 3ms (unchanged)
test/utils/Multicall.t.sol 86ms (unchanged)
test/utils/SignatureChecker.t.sol 64ms (unchanged)
.eslintrc.js 16ms (unchanged)
test/utils/scripts/elements.js 2ms (unchanged)
test/utils/scripts/generate-bad-multiproof-leaves.js 5ms (unchanged)
test/utils/scripts/generate-bad-multiproof-proof-flags.js 3ms (unchanged)
test/utils/scripts/generate-bad-multiproof.js 2ms (unchanged)
test/utils/scripts/generate-bad-proof.js 2ms (unchanged)
test/utils/scripts/generate-leaf.js 1ms (unchanged)
test/utils/scripts/generate-multiproof-leaves.js 2ms (unchanged)
test/utils/scripts/generate-multiproof-proof-flags.js 2ms (unchanged)
test/utils/scripts/generate-multiproof.js 2ms (unchanged)
test/utils/scripts/generate-no-such-leaf.js 2ms (unchanged)
test/utils/scripts/generate-proof-no-such-leaf.js 2ms (unchanged)
test/utils/scripts/generate-proof.js 2ms (unchanged)
test/utils/scripts/generate-root-no-such-leaf.js 1ms (unchanged)
test/utils/scripts/generate-root.js 1ms (unchanged)
test/utils/scripts/multiproof-bad-elements.js 0ms (unchanged)
test/utils/scripts/multiproof-bad-indices.js 1ms (unchanged)
test/utils/scripts/multiproof-indices.js 1ms (unchanged)
.solhint.json 1ms (unchanged)
package.json 0ms (unchanged)
renovate.json 0ms (unchanged)
.github/dependabot.yml 9ms (unchanged)
.github/FUNDING.yml 1ms (unchanged)
.github/ISSUE_TEMPLATE/bug_report.yml 4ms (unchanged)
.github/ISSUE_TEMPLATE/feature_request.yml 2ms (unchanged)
.github/workflows/checks.yml 5ms (unchanged)
.github/workflows/publish-npm.yml 4ms (unchanged)
.github/workflows/publish-pypi.yml 3ms (unchanged)
.github/workflows/publish-test-pypi.yml 3ms (unchanged)
.github/workflows/test-contracts.yml 6ms (unchanged)
.prettierrc.yml 1ms (unchanged)
ape-config.yaml 1ms (unchanged)
pnpm-lock.yaml 126ms (unchanged)
```

**Comment by Janther** [comment]
I'm running:

- macOS `14.2.1`
- pnpm `8.14.1`

**Comment by pcaversaccio** [comment]
@Janther I quickly talked to @fvictorio and he thinks it's a shell issue and how the glob pattern is interpreted. I will let him comment here.

**Comment by pcaversaccio** [comment]
https://medium.com/@jakubsynowiec/you-should-always-quote-your-globs-in-npm-scripts-621887a2a784

**Comment by pcaversaccio** [comment]
TL;DR glob syntax that works across all OSes (at least Windows and Linux) after testing:
- `prettier` =>`'...'`
- `solhint` => `'...'`
- `eslint` => `\"...\"`
- `forge` (`--match-path`)  => `\"...\"`

Conclusion: I hate computers!

**Comment by Janther** [comment]
I'll close this issue as completed then. Feel free to reopen it if something else arises.

**Comment by pcaversaccio** [comment]
> I'll close this issue as completed then. Feel free to reopen it if something else arises.

Sure. So the most stable solution to use across all OSes (i.e. Windows, Linux, Mac) is to use `\"`.
